Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.py

```diff
@@ -100,6 +97,104 @@ def _oproxy_authorize(self) -> Tuple[str, str, str, str, int]:
+            now_datetime = datetime.now()
+            time_from_first_failure = now_datetime - first_failure_datetime
+            time_from_last_failure = now_datetime - last_failure_datetime
+
+            if time_from_first_failure < timedelta(hours=1):
+                if time_from_last_failure < timedelta(minutes=1):
+                    raise DemistoException(err_msg)
+            elif time_from_first_failure < timedelta(hours=48):
+                if time_from_last_failure < timedelta(minutes=10):
+                    raise DemistoException(err_msg)
+            else:
+                if time_from_last_failure < timedelta(minutes=60):
+                    raise DemistoException(err_msg)
+
+        return self._get_access_token()
+
+    def _get_access_token(self) -> dict:
+        """ Performs an http request to oproxy-cdl access token endpoint
+        In case of failure, handles the error, otherwise reset the failure counters and return the response
+
+        Returns: The oproxy response or raising a DemistoException
+
+        """
+        try:
+            oproxy_response = self._http_request('POST',
+                                                 '/cdl-token',
+                                                 json_data={'token': get_encrypted(self.refresh_token, self.enc_key)},
+                                                 timeout=(60 * 3, 60 * 3),
+                                                 retries=3,
+                                                 backoff_factor=10,
+                                                 status_list_to_retry=[400])
+        except DemistoException as e:
+            if self._is_bad_request_error(e):
+                # if the error's status code is 400, update the failure time counters
+                integration_context = demisto.getIntegrationContext()
+                current_time = datetime.now().timestamp()
+                times_dict = {LAST_FAILURE_TIME_CONST: current_time}
+                if not integration_context.get(FIRST_FAILURE_TIME_CONST):
+                    # first failure
+                    times_dict[FIRST_FAILURE_TIME_CONST] = current_time
+                integration_context.update(times_dict)
+                demisto.setIntegrationContext(integration_context)
+            raise e
+
+        self.reset_failure_times()
+        return oproxy_response
+
+    @staticmethod
+    def _is_bad_request_error(demisto_exception: DemistoException):
+        """ Checks if a given error's status code is 400.
+        The first line will be of the following format: 'Error in API call [$STATUS_CODE] - $REASON'
+
+        Args:
+            demisto_exception: the DemistoException that was raised
+
+        Returns: True if the error's status code is 400, False otherwise.
+
+        """
+        err_lines = str(demisto_exception).splitlines()
+        first_line = err_lines[0] if err_lines else ''
+        if '400' in first_line:
+            return True
```

Comment:
Why not just use regex `match` for `Error in API call 400 - .*` instead of splitting and indexing?